staging/src/k8s.io/apiserver/pkg/storage/cacher/cache_watcher.go

```diff
@@ -488,12 +488,6 @@ func (c *cacheWatcher) processInterval(ctx context.Context, cacheInterval *watch
 			break
 		}
 		c.sendWatchCacheEvent(event)
-		// With some events already sent, update resourceVersion so that
-		// events that were buffered and not yet processed won't be delivered
-		// to this watcher second time causing going back in time.
-		//
-		// FIXME: this is bad - it should be maxed for RV=0 case.
-		resourceVersion = event.ResourceVersion
```

Comment:
I'm surprised we're not taking the max of the items in the list like the comment suggested... is there a reason for that?